cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/GitHubHttpHandler.java

```diff
@@ -0,0 +1,453 @@
+      }
+  }
+
+  /**
+   * Deletes current repository info
+   */
+  @Path("repos/github/{repo}")
+  @DELETE
+  public void deleteRepoInfo(HttpRequest request, HttpResponder responder,
+                              @NotNull @PathParam("repo") String repo) throws Exception {
+    if (gitStore.getRepo(repo) != null) {
+      gitStore.deleteRepo(repo);
+      secureStoreManager.delete("system", repo + "_auth_token");
+      responder.sendStatus(HttpResponseStatus.OK);
+    } else {
+      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
+    }
+  }
+
+  @Path("repos/github/{repo}/testconnection")
+  @POST
+  public void testRepoConnection(HttpRequest request, HttpResponder responder,
+      @NotNull @PathParam("repo") String repo) throws Exception {
+    GitHubRepo test = gitStore.getRepo(repo);
+    int responseCode = testRepoConnection(test, new String(secureStore.get("system",
+        repo + "_auth_token").get(), StandardCharsets.UTF_8));
+    if (responseCode == HttpURLConnection.HTTP_OK) {
+      responder.sendString(HttpResponseStatus.OK, "Connection Successful.");
+    } else if (responseCode == HttpURLConnection.HTTP_MOVED_PERM) {
+      responder.sendString(HttpResponseStatus.MOVED_PERMANENTLY, "Repository has been moved.");
+    } else if (responseCode == HttpURLConnection.HTTP_FORBIDDEN) {
+      responder.sendString(HttpResponseStatus.FORBIDDEN, "You do not have access to this repository.");
+    } else {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, "Repository was not found.");
+    }
+  }
+
+  @POST
+  @Path("repos/github/import/{repo}")
+  public void checkInRepo(FullHttpRequest request, HttpResponder responder,
+      @NotNull @PathParam("repo") String repo) throws Exception {
+
+    try {
+      GitHubIO input = GSON
+          .fromJson(request.content().toString(StandardCharsets.UTF_8), GitHubIO.class);
+
+      String branch = !Strings.isNullOrEmpty(input.getBranch()) ?
+          input.getBranch() : gitStore.getRepo(repo).getDefaultBranch();
+
+      String path = input.getPath();
+
+      GitHubRepo gitHubRepo = gitStore.getRepo(repo);
+      URL url = new URL(parseUrl(gitHubRepo.getUrl()) + "/contents/" + path + "?ref=" + branch);
+      HttpURLConnection con = (HttpURLConnection) url.openConnection();
+      con.setRequestMethod("GET");
+      String authString = new String(secureStore.get("system", repo + "_auth_token")
+          .get(), StandardCharsets.UTF_8);
+      con.setRequestProperty("Authorization", authString);
+
+      BufferedReader reader = new BufferedReader(new InputStreamReader(con.getInputStream()));
+
+      if (con.getResponseCode() == HttpURLConnection.HTTP_OK) {
+        responder.sendString(HttpResponseStatus.OK, retrieveContent(reader));
+      } else if (con.getResponseCode() == HttpURLConnection.HTTP_FORBIDDEN) {
+        responder.sendString(HttpResponseStatus.FORBIDDEN, "You do not have access to this file");
+      } else {
+        responder
```

Comment:
why this is NOT FOUND error ? also when you check in a file, why do you require that file to be pre-exist ?

Reply:
Oh, "import" was being used as checkout